packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1093,6 +1092,10 @@ export class RemoteFluidDataStoreContext extends FluidDataStoreContext {
 	private readonly initialSnapshotDetailsP = new LazyPromise<ISnapshotDetails>(async () => {
 		// Sequence number of the snapshot.
 		let sequenceNumber: number | undefined;
+		// Check whether we need to fetch the snapshot first to load.
+		if (this.snapshotFetchRequired === undefined && this._baseSnapshot?.groupId !== undefined) {
+			this.snapshotFetchRequired = this.runtime.isSnapshotFetchRequired([this.id]);
```

Comment:
Why do we reach out to runtime, not a parentContext? I think we should write the code in such a way that it's easier / possible to make this feature work at multiple levels in the future, if possible.